node/cmd/guardiand/adminserver.go

```diff
@@ -342,7 +343,7 @@ func (s *nodePrivilegedService) FindMissingMessages(ctx context.Context, req *no
 	}, nil
 }
 
-func adminServiceRunnable(logger *zap.Logger, socketPath string, injectC chan<- *vaa.VAA, signedInC chan *gossipv1.SignedVAAWithQuorum, db *db.Database, gst *common.GuardianSetState) (supervisor.Runnable, error) {
+func adminServiceRunnable(logger *zap.Logger, socketPath string, injectC chan<- *vaa.VAA, signedInC chan *gossipv1.SignedVAAWithQuorum, obsvReqSendC chan *gossipv1.ObservationRequest, db *db.Database, gst *common.GuardianSetState) (supervisor.Runnable, error) {
```

Comment:
It almost makes sense to use option currying or something to simplify this long function signature down the line.

Reply:
At the very least it's time to split it up into multiple lines, but I didn't want to make the diff noisier than necessary. If we keep adding arguments, we probably want to use some dependency injection framework....

Reply:
Oh sorry I didn't mean for this PR, but for a future cleanup.